ksqldb-common/src/main/java/io/confluent/ksql/logging/processing/MeteredProcessingLogger.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright 2022 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"); you may not use
+ * this file except in compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.logging.processing;
+
+import java.time.Instant;
+import java.util.Map;
+import java.util.Objects;
+import org.apache.kafka.common.MetricName;
+import org.apache.kafka.common.metrics.KafkaMetric;
+import org.apache.kafka.common.metrics.MetricConfig;
+import org.apache.kafka.common.metrics.Metrics;
+import org.apache.kafka.common.metrics.stats.CumulativeSum;
+
+public class MeteredProcessingLogger implements ProcessingLogger {
+  public static final String PROCESSING_LOG_ERROR_METRIC_NAME = "processing-error-total";
+  public static final String PROCESSING_LOG_METRICS_GROUP_NAME = "processing-log-metrics";
+  public static final String PROCESSING_LOG_METRIC_DESCRIPTION =
+      "The total number of errors emitted by the processing log.";
+
+  private final ProcessingLogger logger;
+  private final CumulativeSum sumOfErrors;
```

Comment:
Just checking do we really want to have a cumulative sum (it is ever growing) or a windowed sum?

Reply:
I think we want cumulative sum here because the actual metric we're trying to get is a delta metric, and delta metrics are automatically calculated for cumulative sum ones